ClientSideKeyVaultKeyToCustomerManagedKey/exampleDataCreator/exampleDataCreator.py

```diff
@@ -0,0 +1,110 @@
+import base64
+import os
+from azure.storage.blob import BlobServiceClient
+from azure.keyvault.keys import KeyClient, KeyVaultKey, KeyType
+from azure.identity import ClientSecretCredential
+from azure.keyvault.keys.crypto import CryptographyClient
+from ClientSideKeyVaultKeyToCustomerManagedKey import config as cfg
+from azure.keyvault.secrets import SecretClient
+
+
+def main():
+    # credential required to access client account
+    credentials = ClientSecretCredential(cfg.TENANT_ID, cfg.CLIENT_ID, cfg.CLIENT_SECRET)
+    # access keyvault client using credentials and reference to client keyvault
+    # access blob service client using connection string as reference
+    bs_client = BlobServiceClient.from_connection_string(cfg.connection_str)
+    key_client = KeyClient(vault_url=cfg.KEYVAULT_URL, credential=credentials)
+    secret_client = SecretClient(vault_url=cfg.KEYVAULT_URL, credential=credentials)
+
+    # create or get container
+    try:
+        cont_client = bs_client.create_container(cfg.cont_name)
+    except:
+        cont_client = bs_client.get_container_client(cfg.cont_name)
+
+    # call to methods
+    key_uri = get_key_uri(key_client)
+    create_encryption_scope(key_uri)
+    key_vault_key = get_keyvault_key(key_client, secret_client)
+    content = get_content(cfg.blob_name)
+    upload_blob(cfg.blob_name, content, cont_client, key_vault_key, credentials)
+
+
+class KeyWrapper:
+    # key wrap algorithm for kek
+
+    def __init__(self, kek, credential):
+        self.algorithm = cfg.key_wrap_algorithm
+        self.kek = kek
+        self.kid = kek.id
+        self.client = CryptographyClient(kek, credential)
+
+    def wrap_key(self, key):
+        if self.algorithm != cfg.key_wrap_algorithm:
+            raise ValueError('Unknown key wrap algorithm. {}'.format(self.algorithm))
+        wrapped = self.client.wrap_key(key=key, algorithm=self.algorithm)
+        return wrapped.encrypted_key
+
+    def unwrap_key(self, key, _):
+        if self.algorithm != cfg.key_wrap_algorithm:
+            raise ValueError('Unknown key wrap algorithm. {}'.format(self.algorithm))
+        unwrapped = self.client.unwrap_key(encrypted_key=key, algorithm=self.algorithm)
+        return unwrapped.key
+
+    def get_key_wrap_algorithm(self):
+        return self.algorithm
+
+    def get_kid(self):
+        return self.kid
+
+
+def get_key_uri(k_client):
+    # get key uri from keyvault
+    e_key = k_client.get_key(cfg.serverside_encryption_keyname)
+    k_uri = str(e_key.id)
+
+    return k_uri
+
+
+def create_encryption_scope(k_uri):
```

Comment:
>create_encryption_scope [](start = 4, length = 23)

This should be in migration.py, else exampleDataCreator.py is required to run.

Reply:
I think that we are assuming that the user would already have an encryption scope set up, that they would input the name of into the config.py file, so James, Cecilia, and I put the creation of the encryption scope in the exampleDataCreator, since that file is supposed to demonstrate the entire setup.